### ISSUE TITLE
build error `Implicit declaration of function 'OSAtomicCompareAndSwap32Barrier' is invalid in C99`

### ISSUE THREAD

**Issue by michaelknoch** [flowkey/UIKit-cross-platform#348]
## Current behavior
build error `Implicit declaration of function 'OSAtomicCompareAndSwap32Barrier' is invalid in C99` when building for apple silicon mac. It works when building for Intel mac via rosetta.

## Expected behavior
shuild build for apple silicon macs

## Steps to reproduce
1. build sample mac app for apple silicon

## Screenshots
<img width="1598" alt="Screenshot 2021-11-29 at 10 51 32" src="https://user-images.githubusercontent.com/5617793/143845742-8b3ca33e-f383-467f-8514-4fff75a00436.png">

## Environment
- Device: Mac with M1
- Xcode: 13.1
- OS: 12.0.1
- UIKit Version: master (fc51799c204c871336939ffb203ec45a65b21d0c)

**Comment by ephemer** [comment]
Yep, saw this too. I think we'll need to update SDL to fix it but for now the workaround is to build for rosetta (esp. because building our UIKit for Mac doesn't make a lot of sense since it's possible to build real UIKit apps for Apple Silicon macs)

**Comment by ephemer** [comment]
Closing this because it's actually been fixed for at least a year
